### DISCUSSION TITLE
Move a composed resource into a differnt composition without deleting external resource

### DISCUSSION THREAD

**Discussion by twobiers** [crossplane/crossplane#6114] [Q&A]
I'm currently thinking about a way of transfering composed resources around. In my use case I have a composition creating a kubernetes cluster and a subnet (lets call it `KubernetesCluster`), and a differnt one creating a virtual network (`VirtualNetwork`).

Now I realized a mistake during the design phase and want to make the subnet part rather of the `VirtualNetwork` composition.

I think I could realize that using the following steps:
1. Apply the updated VirtualNetwork composition
1. Let it try to create a subnet which will fail in the external provider.
1. Set the `crossplane.io/external-name` annotation with the existing name in the provider
1. Let it resolve
1. Under the existing `KubernetesCluster` composite resource update the already composed subnet with a `managementPolicy` that prevents deletions.
1. Remove the Subnet from the `KubernetesCluster` composition and apply it to the cluster

Is this the preferred way to do this or is  there a better way to achieve this?

**Comment by jbw976** [comment]
Is this issue about refactoring compositions helpful for your scenario too @twobiers? 

* https://github.com/crossplane/crossplane/issues/5654

**Reply by twobiers** [reply]
Looks promising to me
